src/mscorlib/src/System/Resources/ResourceReader.cs

```diff
@@ -1373,8 +1373,9 @@ public bool MoveNext()
 
                     String key;
                     Object value = null;
+                    lock (_reader) { // locks should be taken in the same order as in RuntimeResourceSet.GetObject to avoid deadlock
                     lock (_reader._resCache) {
```

Comment:
@ramarag, sorry for broken indentation, just tried to keep number of lines changed at minimum.
I added an indentation and rebased new commit to master). Please review.